BnbnavNetClient/Models/Player.cs

```diff
@@ -11,9 +28,130 @@ public sealed class Player
     public Edge? SnappedEdge { get; private set; }
 
     public double MarkerAngle { get; private set; }
+    
+    public FormattedText PlayerText { get; }
+    
+    public Point MarkerCoordinates {
+        get
+        {
+            if (SnappedEdge is null) return new(X, Z);
+            
+            //Find the intersection point
+            var playerLine = new ExtendedLine()
+            {
+                Point1 = new(X, Z),
+                Point2 = new(X + 1, Z)
+            };
+            playerLine = playerLine.SetAngle(SnappedEdge.Line.NormalLine().Angle);
+            _ = playerLine.TryIntersect(SnappedEdge.Line, out var intersectionPoint);
+            return intersectionPoint;
+        }
+    }
+
+    public List<(DateTime, Point)> PosHistory { get; set; } = new();
+
+    public ExtendedLine Velocity => new()
+    {
+        Point1 = PosHistory.Last().Item2,
+        Point2 = PosHistory.First().Item2
+    };
 
-    public Player(string name)
+    public Player(string name, MapService mapService)
     {
+        _mapService = mapService;
         Name = name;
+        
+        PlayerText = new(Name, CultureInfo.CurrentCulture, FlowDirection.LeftToRight,
+            new("Noto Sans"), 20, null);
+
+        _timer = new(50);
+        _timer.Elapsed += (_, _) =>
```

Comment:
i'm a little confused about the timer stuff, aren't we getting constant position ticks from the server?

Reply:
Position updates from the server can come at any time - this timer ensures that the angular velocity of the player does not depend on network conditions.

Reply:
Perhaps the timer should be in MapView instead?

Reply:
Angle of a player is a property of the player, not the map view - if we ever have two map views all the players will turn at double speed

Reply:
some, of the logic should probably be done in mapview, as we only really need to check snapped edges and angular velocity of rendered players. it would probably help improve perforamcne

Reply:
Resolving for now - we can come back to this later